kernel/src/common/take_cell.rs

```diff
@@ -86,18 +88,20 @@ impl<'a, T: ?Sized> TakeCell<'a, T> {
     /// # Examples
     ///
     /// ```
-    /// let cell = TakeCell::new(1234);
+    /// use kernel::common::cells::TakeCell;
+    /// let mut value = 1234;
+    /// let cell = TakeCell::new(&mut value);
     /// let x = &cell;
     /// let y = &cell;
     ///
     /// x.map(|value| {
     ///     // We have mutable access to the value while in the closure
-    ///     value += 1;
+    ///     *value += 1;
     /// });
     ///
     /// // After the closure completes, the mutable memory is still in the cell,
     /// // but potentially changed.
-    /// assert_eq!(y.take(), Some(1235));
+    /// assert_eq!(y.take(), Some(&mut 1235));
```

Comment:
This isn't actually valid code, right?

Reply:
I believe it is, since `&mut` implements `PartialEq`.

Reply:
I take that back, I'm not 100% sure and it's worth testing. The pointer value will be different, since this code is going to allocate a _new_ value (`12345`) on the stack and take a reference to it, which will have a different pointer value. It basically comes down to whether the `PartialEq` implementation for `&mut` checks equality of the referenced values or of the references themselves.

Reply:
Apparently this does indeed work (so the `PartialEq` implementation compares values). In rust playground, the following code prints "Success" and does not have an assertion.

```
    let mut val = 1234;
    val += 1;
    assert_eq!(&mut val, &mut 1235);
    println!("Success");
```

Reply:
`cargo test` ran the doc tests, which exercised this. Initially, I had it without the `&mut` on `&mut 1235` and it failed; add `&mut` was the secret sauce to make it go.